modules/tealium_adobevisitor/ios/Classes/SwiftTealiumAdobevisitorPlugin.swift

```diff
@@ -0,0 +1,143 @@
+        
+        if let adobeVisitorOrgId = adobeVisitorOrgId {
+            config.adobeVisitorOrgId = adobeVisitorOrgId
+        }
+        if let adobeVisitorExistingEcid = adobeVisitorExistingEcid {
+            config.adobeVisitorExistingEcid = adobeVisitorExistingEcid
+        }
+        if let adobeVisitorRetries = adobeVisitorRetries {
+            config.adobeVisitorRetries = adobeVisitorRetries
+        }
+        if let adobeVisitorAuthState = adobeVisitorAuthState,
+           let adobeVisitorAuthState = AdobeVisitorAuthState(rawValue: adobeVisitorAuthState) {
+            config.adobeVisitorAuthState = adobeVisitorAuthState
+        }
+        if let adobeVisitorDataProviderId = adobeVisitorDataProviderId {
+            config.adobeVisitorDataProviderId = adobeVisitorDataProviderId
+        }
+        if let adobeVisitorCustomVisitorId = adobeVisitorCustomVisitorId {
+            config.adobeVisitorCustomVisitorId = adobeVisitorCustomVisitorId
+        }
+    }
+    
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        if call.method == "configure" {
+            configure(call, result: result);
+        } else if call.method == "linkEcidToKnownIdentifier" {
+            linkExistingEcidToKnownIdentifier(call, result: result)
+        } else if call.method == "getAdobeVisitor" {
+            getAdobeVisitor(result)
+        } else if call.method == "resetVisitor" {
+            resetVisitor()
+        } else if call.method == "decorateUrl" {
+            decorateUrl(call, result: result)
+        } else {
+            result(false)
+        }
+    }
+    
+    func configure(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        guard let arguments = call.arguments as? [String: Any] else {
+            return result(false)
+        }
+        
+        if let adobeVisitorOrgId = arguments["adobeVisitorOrgId"] as? String {
+            self.adobeVisitorOrgId = adobeVisitorOrgId
+        }
+        if let adobeVisitorExistingEcid = arguments["adobeVisitorExistingEcid"] as? String {
+            self.adobeVisitorExistingEcid = adobeVisitorExistingEcid
+        }
+        if let adobeVisitorRetries = arguments["adobeVisitorRetries"] as? Int {
+            self.adobeVisitorRetries = adobeVisitorRetries
+        }
+        if let adobeVisitorAuthState = arguments["adobeVisitorAuthState"] as? Int {
+            self.adobeVisitorAuthState = adobeVisitorAuthState
+        }
+        if let adobeVisitorDataProviderId = arguments["adobeVisitorDataProviderId"] as? String {
+            self.adobeVisitorDataProviderId = adobeVisitorDataProviderId
+        }
+        if let adobeVisitorCustomVisitorId = arguments["adobeVisitorCustomVisitorId"] as? String {
+            self.adobeVisitorCustomVisitorId = adobeVisitorCustomVisitorId
+        }
+    }
+    
+    func getAdobeVisitor(_ result: @escaping FlutterResult) {
+        result(adobeModule?.visitor?.asDictionary())
```

Comment:
I believe since the visitor class is a Codable there is an `encoded` property that will give you the same result